### ISSUE TITLE
[Build] [CMake]More targets need to output.

### ISSUE THREAD

**Issue by QiuYilin** [microsoft/onnxruntime#18872]
### Describe the issue

From https://github.com/microsoft/onnxruntime/pull/8919 ,find_package is supported. But it only contains target onnxruntime which can work on CPU. If I use cuda, onnxruntime_providers_shared.dll and onnxruntime_providers_cuda.dll will not be located. 

The generated onnxruntimeTargets-release.cmake
```
#----------------------------------------------------------------
# Generated CMake target import file for configuration "Release".
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Import target "onnxruntime::onnxruntime" for configuration "Release"
set_property(TARGET onnxruntime::onnxruntime APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(onnxruntime::onnxruntime PROPERTIES
  IMPORTED_IMPLIB_RELEASE "${_IMPORT_PREFIX}/lib/onnxruntime.lib"
  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/bin/onnxruntime.dll"
  )

list(APPEND _cmake_import_check_targets onnxruntime::onnxruntime )
list(APPEND _cmake_import_check_files_for_onnxruntime::onnxruntime "${_IMPORT_PREFIX}/lib/onnxruntime.lib" "${_IMPORT_PREFIX}/bin/onnxruntime.dll" )

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
```

### Target platform

Windows

### Build script

`.\build.bat --config Release --build_shared_lib --parallel --compile_no_warning_as_error --skip_submodule_sync`

### Error / output

Only target onnxruntime is ouput.

**Comment by diablodale** [comment]
Agreed, the ORT v16 "cmake find" support is incomplete...it does not have the runtime dependencies.

* missing providers `onnxruntime_providers_shared.dll`, `onnxruntime_providers_cuda.dll`, `onnxruntime_providers_tensorrt.dll`, ...
* missing dependencies of each those providers, e.g. dml needs `DirectML.dll`, ...

I continue to use my own `cmake\onnxruntime_providers.cmake` and `Findonnxruntime.cmake`.

* my `onnxruntime_providers.cmake` updates the install commands for providers I need so the cmake install will correctly copy all runtime dependencies (provider dlls, directml.dll, etc.) into the install directories. It also copies missing includes that I need (which is already being discussed across several other issues so I'll not discuss that here)
* my `Findonnxruntime.cmake` uses the `find_package(onnxruntime REQUIRED COMPONENTS onnxruntime_providers_<PROVIDER_NAME> ...)` components approach to probe `if(EXISTS "${ORT_RUNTIME_DIR}onnxruntime_providers_${provider_name}.dll")` the install dir for any number of providers and then imports that MODULE `add_library(onnxruntime::onnxruntime_providers_${provider_name} MODULE IMPORTED)` and sets its properties for IMPORTED_CONFIGURATIONS, INTERFACE_INCLUDE_DIRECTORIES, IMPORTED_LOCATION.

Naturally, this is a probing and importing approach for cmake projects like ORT v16 who do not expose their components and runtime dependencies. The newest cmake versions now support (probably) all that is needed to do such through the new PUBLIC_HEADER, RUNTIME_DEPENDENCY/_SET, IMPORTED_RUNTIME_ARTIFACTS , etc.

Using the newest native RUNTIME_DEPENDENCY, ARTIFACTS, etc. is probably a good strategic approach. But it requires every provider to go through all its dependency needs and ensure they are transitively exported for install and cmake.

In constrast, probing might be faster to code for specific providers (like I have done) when there is only a single provider file (e.g. DLL), maybe a single transitive dependency (e.g. DirectML.DLL)
